pkg/autoops/api/operation.go

```diff
@@ -86,15 +98,27 @@ func disableFeature(
 		EnvironmentNamespace: environmentNamespace,
 	}
 	_, err := featureClient.DisableFeature(ctx, req)
-	if code := status.Code(err); code == codes.FailedPrecondition {
-		logger.Warn(
-			"Failed to disable feature",
+	if err != nil {
+		if code := status.Code(err); code == codes.FailedPrecondition {
```

Comment:
I split the implementation to print warning and error levels to make debugging easier.